### PR TITLE
Introduce snmp_exporter integration

### LINKED ISSUE #1370: SNMP exporter integration
This issue to track [snmp_exporter](https://github.com/prometheus/snmp_exporter) integration. We would like to add in order to deliver multiple SNMP based integrations later on.

There is #1364 PR draft with work in progress.

### PR DESCRIPTION
**Pull request by v-zhuravlev**
#### PR Description 
This integration adds [snmp_exporter](https://github.com/prometheus/snmp_exporter). Closes #1370. 

#### Notes to the Reviewer

1) On top of just embedding snmp_exporter this implements `walk_params` that can help to provide custom connection/auth SNMP settings.
2) snmp_exporter module in go.mod points to specific commit from master branch, as the latest snmp_exporter release didn't have snmp collector logic  as the separate go module

#### PR Checklist

This checklist describes expected features/modes of operations.

- [x] Implement snmp_exporter in a way, that it can be used w/o relabelling configs and with autoscrape support (see #1367 ). 
This mode could be great and user-friendly for trivial 1-2 SNMP devices integrations that can be provided statically
- [x] Implement snmp_exporter in a way, that it's still possible to scrape snmp_targets with multi-target approach (https://prometheus.io/docs/guides/multi-target-exporter/):

This mode could be great when we need to scrape dynamic or large fleet of network devices and so we can use prometheus supported service discovery capabilities (dns_sd, file_sd, consul_sd etc)

```yaml 
metrics:
  wal_directory: /tmp/wal
  configs:
    - name: snmp_targets
      scrape_configs:
        - job_name: 'snmp'
          static_configs:
            - targets: [192.168.15.1]  # First SNMP device.
              labels:
                __param_walk_params: public
            - targets: [192.168.15.2]  # Second SNMP device.
              labels:
                __param_walk_params: private
          metrics_path: /integrations/snmp_exporter/snmp
          params:
            module: [if_mib] # snmp module to use
          relabel_configs:
            - source_labels: [__address__]
              target_label: __param_target
            - source_labels: [__param_target]
              target_label: instance
            - replacement: 127.0.0.1:9090  # The SNMP exporter's real hostname:port.
              target_label: __address__ 
metrics:
  wal_directory: /tmp/wal
  configs:
    - name: snmp_targets
      scrape_configs:
        - job_name: 'snmp'
          static_configs:
            - targets: [192.168.15.1]  # First SNMP device.
              labels:
                __param_walk_params: public
            - targets: [192.168.15.2]  # Second SNMP device.
              labels:
                __param_walk_params: private
          metrics_path: /integrations/snmp_exporter/metrics
          params:
            module: [if_mib] # snmp module to use
          relabel_configs:
            - source_labels: [__address__]
              target_label: __param_target
            - source_labels: [__param_target]
              target_label: instance
            - replacement: 127.0.0.1:9090  # The SNMP exporter's real hostname:port.
              target_label: __address__ 
integrations:
  snmp_exporter:
    enabled: true
    scrape_integration: false
    walk_params:
      private:
        version: 2
        auth:
          community: secretpassword
      public:
        version: 2
        auth:
          community: public
server:
    http_listen_port: 9090
```

- [ ] Implement endpoint (/integrations/snmp_exporter/metrics) for SNMP exporter own metrics with autoscrape.
- [x] Add support for walk_params HTTP param, that can be used to override SNMP auth and other connection settings for any specific device(s) 
- [x] Embed snmp.yml into exporter from upstream project
- [x] Add support for custom snmp.yml to be provided by end users
- [x] CHANGELOG updated 
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/user/configuration/integrations/integrations-next/snmp-config.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
++++
+title = "SNMP Integration"
++++
+
+# snmp config
+
+The `snmp` block configures the `snmp` integration,
+which is an embedded version of
+[`snmp_exporter`](https://github.com/prometheus/snmp-exporter). This allows collection of SNMP metrics from the network devices with ease.
```
Comment (tpaschalis) [inline docs/user/configuration/integrations/integrations-next/snmp-config.md:9]: The link with the `-` is a 404.
```suggestion
[`snmp_exporter`](https://github.com/prometheus/snmp_exporter). This allows collection of SNMP metrics from the network devices with ease. 
```

Comment (v-zhuravlev) [reply docs/user/configuration/integrations/integrations-next/snmp-config.md]: thanks

**Comment by tpaschalis** [comment]
cc @rfratto @v-zhuravlev I've added e927cd6 which uses a single copy of snmp.yml, compressed using GZIP and a new go:generate. 

Before this change, the new exporter plus `go:embed` directive were inflating the binary size by 3MB, now it's 612KB.
```
$ du -k agent.*
156420	agent.new
158852	agent.old
155808	agent.orig
```

File: docs/user/configuration/integrations/integrations-next/snmp-config.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  extra_labels:
+    [ <labelname>: <labelvalue> ... ]
+
+  #
+  # Exporter-specific configuration options
+  #
+
+  # SNMP configuration file with custom modules.
+  # See https://github.com/prometheus/snmp_exporter#generating-configuration for more details how to generate custom snmp.yml file. 
+  # If not defined, embedded snmp_exporter default set of modules is used.
```
Comment (mattdurham) [inline docs/user/configuration/integrations/integrations-next/snmp-config.md:117]: Link to embedded file?

**Comment by tpaschalis** [comment]
@mattdurham do you think we're all set to merge this?

**Comment by mattdurham** [comment]
Ship it

**Pull request merged**